addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,340 @@
-        return str(device.idVendor) + ":" + str(device.idProduct)
-
-    def get_connection(self):
-        return 'direct'
-
-    def value(self):
-        return self.value
 
+class StatusController(http.Controller):
+    @http.route('/hw_drivers/action', type='json', auth='none', cors='*', csrf=False)
+    def action(self, device_id, data):
+        if device_id in iot_devices:
+            return iot_devices[device_id].action(data)
+        else:
+            return {'message': _('Device %s not found') % device_id}
+
+    @http.route('/hw_drivers/event', type='json', auth='none', cors='*', csrf=False)
+    def event(self, requests):
+        req = DeviceManager.addRequest(requests)
+        if req['event'].wait(58):
+            req['event'].clear()
+            return_value = req['queue'].copy()
+            return return_value
+
+    @http.route('/hw_drivers/box/connect', type='http', auth='none', cors='*', csrf=False)
+    def connect_box(self, token):
+        """
+        This route is called when we want that a IoT Box will be connected to a Odoo DB
+        token is a base 64 encoded string and have 2 argument separate by |
+        1 - url of odoo DB
+        2 - token. This token will be compared to the token of Odoo. He have 1 hour lifetime
+        """
+        server = get_odoo_server_url()
+        if server:
+            return _('This IoTBox has already been connected')
+        else:
+            iotname = ''
+            token = b64decode(token).decode('utf-8')
+            url = token.split('|')[0]
+            token = token.split('|')[1]
+            reboot = 'noreboot'
+            path = os.getenv("HOME") + '/odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh'
+            subprocess.call([path, url, iotname, token, reboot])
+            m.send_alldevices()
+            return _('IoTBox connected')
 
 #----------------------------------------------------------
-# Bluetooth
+# Drivers
 #----------------------------------------------------------
-class GattBtManager(gatt.DeviceManager):
-
-    def device_discovered(self, device):
-        # TODO: need some kind of updated_devices mechanism or not?
-        path = "bt_%s" % (device.mac_address,)
-        if path not in drivers:
-            for driverclass in btdrivers:
-                d = driverclass(device = device, manager=self)
-                if d.supported():
-                    drivers[path] = d
-                    d.connect()
-                    send_iot_box_device(False)
-
-
-class BtManager(Thread):
-    gatt_manager = False
-
-    def run(self):
-        dm = GattBtManager(adapter_name='hci0')
-        self.gatt_manager = dm
-        dm.start_discovery()
-        dm.run()
 
-#----------------------------------------------------------
-# Bluetooth drivers
-#----------------------------------------------------------
-btdrivers = []
+drivers = []
+bt_devices = {}
+iot_devices = {}
 
-class BtMetaClass(type):
+class MetaClass(type):
```

Comment:
rename into DriverMetaClass

Reply:
add a docstring to this class ; explain that it's just a hook to register the driver into the `drivers` list